compatibility_server/pip_checker.py

```diff
@@ -353,24 +356,44 @@ def _build_command(self, subcommands: List[str]):
 
     def _install(self, container: docker.models.containers.Container):
         """Run pip install in the container."""
-        command = self._build_command(['install', '-U'] + self._packages)
-        returncode, output = self._run_command(
-            container,
-            command,
-            stdout=False,
-            stderr=True,
-            raise_on_failure=False)
-        if returncode:
-            # Checking for environment error
-            environment_error = PIP_ENVIRONMENT_ERROR_PATTERN.search(output)
-            if environment_error:
-                raise PipError(error_msg=environment_error.group('error'),
-                               command=command,
-                               returncode=returncode)
-
-            return PipCheckResult(self._packages,
-                                  PipCheckResultType.INSTALL_ERROR,
-                                  output)
+        # If there is cloud client libraries, install it from source.
+        # Else install it from PyPI.
+        for pkg in self._packages:
+            if GITHUB_CLIENTLIBS_PREFIX in pkg:
+                # Shallow clone the repository
+                shallow_clone_command = ['git', 'clone', '--depth', '1',
+                                         CLIENTLIBS_GITHUB_URL]
+                self._run_command(
```

Comment:
I don't like that we can run this multiple times and every time after the first will fail. I also don't like that we don't check for failures.